includes/class-wc-order-item-product.php

```diff
@@ -375,7 +375,7 @@ public function get_item_downloads() {
 					$files[ $download_id ]['download_url']        = add_query_arg( array(
 						'download_file' => $product_id,
 						'order'         => $order->get_order_key(),
-						'email'         => urlencode( $order->get_billing_email() ),
+						'uid'         	=> hash( 'sha256', $order->get_billing_email() ),
```

Comment:
We could just change to MD5 as Mike said.

Reply:
I like that idea in terms of performance, but I'm consistently reading that Google requires at least SHA-256 for even one-way hashing, especially considering the possible vulnerabilities of `md5`. 

https://developers.google.com/analytics/solutions/crm-integration#user_id
http://searchsecurity.techtarget.com/news/4500270165/MD5-vulnerability-renews-calls-for-faster-SHA-256-transition

It looks like WordFence has also acknowledged the possible insecurity of using `md5` for hashing: 
https://www.wordfence.com/blog/2016/12/gravatar-advisory-protect-email-address-identity/

Reply:
I see that this is your personal preference, since we don't need to consider vulnerabilities, since today we already display the email address anyway, we are changing to a hash just to hide this information from Google.